pkgs/applications/virtualization/vmware-workstation/default.nix

```
@@ -0,0 +1,419 @@
+{ stdenv, buildFHSUserEnv, fetchurl, lib, zlib, gdbm, bzip2,
+libxslt, libxml2, libuuid, readline, xz, cups, glibc, libaio,
+vulkan-loader, alsa-lib, libpulseaudio, libGL, numactl, libX11,
+libXi, kmod, python3, autoPatchelfHook, makeWrapper, sqlite,
+enableInstaller ? false }:
+
+with lib;
+
+let
+  vmware-unpack-env = buildFHSUserEnv rec {
+    name = "vmware-unpack-env";
+    targetPkgs = pkgs: [ zlib ];
+  };
+  gdbm3 = gdbm.overrideAttrs ( old: rec {
+    version = "1.8.3";
+
+    src = fetchurl {
+      url = "mirror://gnu/gdbm/${old.pname}-${version}.tar.gz";
+      sha256 = "sha256-zDQDOKLii0AFirnrU1SiHVP4ihWC6iG6C7GFw3ooHck=";
+    };
+
+    installPhase = ''
+      mkdir -p $out/lib
+      cp .libs/libgdbm*.so* $out/lib/
+    '';
+
+  });
+in stdenv.mkDerivation rec {
+  pname = "vmware-workstation";
+  version = "16.2.3";
+  build = "19376536";
+
+  buildInputs = [ libxslt libxml2 libuuid gdbm3 readline xz bzip2 cups glibc libaio vulkan-loader alsa-lib libpulseaudio libGL numactl libX11 libXi kmod ];
+  nativeBuildInputs = [ python3 vmware-unpack-env autoPatchelfHook makeWrapper ]
+  ++ lib.optional (enableInstaller) sqlite;
+
+  src = fetchurl {
+    url = "https://download3.vmware.com/software/WKST-1623-LX-New/VMware-Workstation-Full-${version}-${build}.x86_64.bundle";
+    sha256 = "sha256-+JE1KnRfawcaBannIyEr1TNZTF7YXRYYaFMVq0/erbM=";
+  };
+
+  unpackPhase = ''
+    ${vmware-unpack-env}/bin/vmware-unpack-env -c "sh ${src} --extract unpacked"
+  '';
+
+  installPhase = ''
+
+    mkdir -p \
+      $out/bin \
+      $out/etc/vmware \
+      $out/etc/init.d \
+      $out/lib/vmware \
+      $out/share/doc
+
+    #### Replicate vmware-installer's order but VMX first because of appLoader
+
+    ${lib.optionalString (enableInstaller) ''
+      ## VMware installer
+      echo "Installing VMware Installer"
+      unpacked="unpacked/vmware-installer"
+
+      local vmware_installer_version=$(cat "unpacked/vmware-installer/manifest.xml" | grep -oPm1 "(?<=<version>)[^<]+")
```

```suggestion
      vmware_installer_version=$(cat "unpacked/vmware-installer/manifest.xml" | grep -oPm1 "(?<=<version>)[^<]+")
```
Variables can't really bleed into other things so we could leave that out.